ga4gh/datamodel/variants.py

```diff
@@ -490,9 +506,42 @@ def _addFile(self, filename):
                 if chrom in self._chromFileMap:
                     raise Exception("cannot have overlapping VCF/BCF files.")
                 self._updateMetadata(varFile)
+                self._updateCallSetIds(varFile)
                 self._chromFileMap[chrom] = varFile
 
-    def convertVariant(self, record):
+    def convertVariant(self, record, callSetIds):
+        """
+        Converts the specified pysam variant record into a GA4GH Variant
+        object. Only calls for the specified list of callSetIds will
+        be included.
+        """
+        def _convertGaCall(recordId, recordInfo, call):
+            c = protocol.GACall()
+            """
+            callSetId
+            callSetName
+            genotype
+            phaseset
+            genotypeLikelihood
+            info
+            """
+            c.callSetId = recordId
+            c.callSetName = name
+            # c.genotype = _encodeValue(call.allele_indices)
+            # QUESTION: do not encode? as shown on
+            # http://ga4gh.org/documentation/api/v0.5.1/
+            # ga4gh_api.html#/schema/org.ga4gh.GACall
+            # says the genotype type is array of int?
+            c.genotype = list(call.allele_indices)
+            # TODO: phaseset is not set by pysam
+            c.phaseset = None
+            c.genotypeLikelihood = []
+            for key, value in call.iteritems():
+                if key == 'GL' and value is not None:
+                    c.genotypeLikelihood = [v for v in value]
+            c.info = recordInfo
```

Comment:
This is a bug I think: are you adding all of the info fields for the variant into the Call? The call.info field is for info tags on the call itself. All you'll have here is something like:
```python
for key, value in call.iteritems():
    if key == 'GL':
        if value is not None:
            c.genotypeLikelihood = list(value)
    else:
          c.info[key] = _encodeValue(value)
```